### PR TITLE
Longw/containerlogv2  windows

### PR DESCRIPTION
**Pull request by wanlonghenry**
Longw/containerlogv2 windows fetch

### CODE REVIEW

File: source/plugins/go/src/ingestion_token_utils.go [hunk 1]
```diff
@@ -425,6 +430,8 @@ func getAgentConfiguration(imdsAccessToken string) (configurationId string, chan
 		SendException(message)
 		return configurationId, channelId, err
 	}
+	Log("longwtest1: %v", agentConfiguration)
+	Log("longwtest2: %v", string(responseBytes))
```
Comment (ganga1980) [inline source/plugins/go/src/ingestion_token_utils.go:434]: remove these log messages after your test.

File: source/plugins/go/src/ingestion_token_utils.go [hunk 2]
```diff
@@ -442,7 +449,12 @@ func getAgentConfiguration(imdsAccessToken string) (configurationId string, chan
 
 	configurationId = agentConfiguration.Configurations[0].Configurationid
 	channelId = agentConfiguration.Configurations[0].Content.Channels[0].ID
-
+	if len(agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights) > 0 {
+		// Assuming you have an instance of AgentConfiguration named agentConfiguration
+		ContainerLogSchemaV2 = agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights[0].Extensionsettings.DataCollectionSettings.EnableContainerLogV2
```
Comment (ganga1980) [inline source/plugins/go/src/ingestion_token_utils.go:454]: ExtensionSettings can be empty and  add  empty/nil check.

File: source/plugins/go/src/ingestion_token_utils.go [hunk 3]
```diff
@@ -442,7 +449,12 @@ func getAgentConfiguration(imdsAccessToken string) (configurationId string, chan
...
+	}
+	Log("longwtest3: %v", ContainerLogSchemaV2)
```
Comment (ganga1980) [inline source/plugins/go/src/ingestion_token_utils.go:456]: remove this test log message

File: source/plugins/go/src/ingestion_token_utils.go [hunk 4]
```diff
@@ -78,6 +78,11 @@ type AgentConfiguration struct {
 				Containerinsights []struct {
 					ID            string   `json:"id"`
 					Originids     []string `json:"originIds"`
+					Extensionsettings struct {
+						DataCollectionSettings struct {
+							EnableContainerLogV2 bool `json:"enableContainerLogV2"`
```
Comment (ganga1980) [inline source/plugins/go/src/ingestion_token_utils.go:83]: This struct other fields as well, right? Lets keep this struct consistent with other data collection settings as well.

**Comment by wanlonghenry** [comment]
image

**Comment by wanlonghenry** [comment]
image

File: source/plugins/go/src/ingestion_token_utils.go (hunk 4 above)
Comment (wanlonghenry) [reply source/plugins/go/src/ingestion_token_utils.go]: added

File: source/plugins/go/src/ingestion_token_utils.go (hunk 2 above)
Comment (wanlonghenry) [reply source/plugins/go/src/ingestion_token_utils.go:456]: Extensionsettings and DataCollectionSettings are embedded structs, not pointers to structs or slices/arrays. In Go, struct fields are automatically initialized to their zero values
so if not set, the output of agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights[0].Extensionsettings will be { { "" "" nil false } }

**Comment by wanlonghenry** [comment]
image

**Pull request merged**